helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -460,6 +460,7 @@ private void globalRebalance(ResourceControllerDataProvider clusterData, Map<Str
           doGlobalRebalance(clusterData, resourceMap, algorithm, currentStateOutput, !waitForGlobalRebalance,
               clusterChanges);
         } catch (HelixRebalanceException e) {
+          _rebalanceFailureCount.increment(1L);
```

Comment:
For my own understanding, how was it working in the past? Did we simply skip the error and not counting?

Reply:
Yes, Global Rebalance failure was never accounted for in the failure metric. See #2062 .